configs/components/echam/echam.yaml

```diff
@@ -391,26 +392,41 @@ choose_scenario:
                         swflux: histswflux
                         MAC-SP: MAC-SP
 
-        # Shared Socioeconomic Pathways (SSPs) scenarios. Forcing data are given
-        # under the choose_scenario_type block
-        ssp585:
-                scenario_type: cmip6
-
-        ssp370:
-                scenario_type: cmip6
-
-        ssp126:
-                scenario_type: cmip6
-
-        ssp245:
-                scenario_type: cmip6
-
-        ssp534os:
-                scenario_type: cmip6
-
         cmip6hist:
                 scenario_type: cmip6
 
+# Shared Socioeconomic Pathways (SSPs) scenarios. Forcing data are given
+# under the choose_scenario_type block
+scenario_is_ssp: "$(( 'ssp' in '${echam.scenario}' ))"
+choose_scenario_is_ssp:
+    True:
+        scenario_type: cmip6
+        data_set_not_r0008: "$(( '${dataset}' != 'r0008' ))"
+        choose_data_set_not_r0008:
+            True:
+                add_warning:
+                    "echam dataset":
+                        message: "
+                            You are trying to run a ``ssp`` simulation with the
+                            ``echam.dataset: ${dataset}``. Some input files might be
+                            missing for ssp simulations in that dataset. It's
+                            recommended that you use the ``r0008`` dataset instead.\n\n
+                            If you wish to continue with the current dataset answer
+                            ``Y`` to the question. If you want to switch to
+                            ``r0008`` dataset, answer ``n`` and modify your runscript
+                            accordingly (``echam.dataset: r0008``)"
+                        ask_user_to_continue: True
+        after_2014: "$(( ${start_date!year} > 2014 ))" # TODO: double check with Sebastian, Antonio, Pengyang, Tido
+        choose_after_2014:
```

Comment:
I think the idea here, is that they are all "flavors of CMIP6" correct?

Reply:
I am not sure if I understand your question. There are a lot of different cmip6-experiments. I would not opt for capturing those by the esm_tools. I like the namelist-approach more.

Reply:
I see what you mean. On the other hand, all this logic with the scenario-type was developed by the FOCI (ECHAM+NEMO) developers some time ago, and has greatly simplified the echam yamls and datasets and make them less error-prone. The idea behind it is that if you select `echam.scenario: ssp585` for example, and your start date is for example 2000, it starts with historical, and then switches to ssp input in 2014. Additionally, if you just want to run a historical simulation, then you just need to select `echam.scenario: cmip6hist`. The `scenario_type` is set automatically in this yaml file for you, and for ssp scenarios or cmip6hist the value that it takes automatically is `cmip6`, which I agree, it's maybe not the most accurate name.

That been said, the ssp logic was developed first in ESM-Tools by people in GEOMAR some time ago, I don't feel like removing that logic, specially because it works pretty well (in fact, I believe this is already part of your ssp simulations configuration correct?)

Reply:
On the other hand, it would be very easy to change the `namelist_dir` here instead of changing `iaero` with a `namelist_changes` and that won't give problems to anyone.

The nice thing about having it controlled through a namelist_changes instead of the namelist, is that if you define your own `namelist_dir` in the runscript, the namelist changes takes care of switching iaero once it reaches 2014, otherwise, you'll still run with whatever your original namelist was.

Reply:
Yes I see the advantages. However, I think the less complex the better. And 
> your start date is for example 2000, it starts with historical, and then switches to ssp input in 2014
means that the namelist will change within one experiment? I would vote absolutely against this. This makes debugging, analysis, etc. way more difficult since it removes the assumption that namelists (and similar stuff) are constant for one experiment (read: a directory). But jeah, keep it as it is if that is the community concensus.

Reply:
IMO this is totally up to the users, I really don't have a strong opinion in favor of one or another, and I am willing to change whatever is needed. I agree  automating everything is not always a good idea. However, there are some users already using this feature (or worse, not having this feature, having gone within the same simulation from a historical scenario to a ssp without changing the iaero), and it seems it's something the MPI-ESM scripts also do (please, @christian-stepanek correct me if I'm wrong).

Reply:
> This makes debugging, analysis, etc. way more difficult since it removes the assumption that namelists

I don't agree with this, all namelist for all runs are saved as `<base_dir>/<exp_id>/config/echam/namelist.echam_<timestamp>`.

Reply:
@christian-stepanek , @PengyangSong , @pgierz , @seb-wahl , can we make a decision here to the question:

- Do we want automatic switching of iaero for simulations that go from historical to ssp times?

Reply:
This may sound trivial but "the user should get the run she/he expected to happen". If I say SSP 585 then that should just be what is done. Or is that not what is currently happening? If not, why not, and how do I help to get that to work?

Reply:
What is the "official" policy on aerosols?

Reply:
IMHO a historical simulation should end when the historical forcing ends. So a break at the end of 2014 would be the expected behavior. If somebody wants to run the simulation longer, then they should explicitly think about under which conditions the simulation should continue - and that somehow implies that the esm-tools interrupt the simulation at 2014.

Again, this is just my point of view.

Reply:
Notice if we have enough input files for year 2014, because it requires some forcing files at 2015.

Reply:
> Notice if we have enough input files for year 2014, because it requires some forcing files at 2015.

Yes but thats a purely technical problem.

Reply:
I don't think the question of Pengyang it's purely a technical problem. We can all code a solution for that in the yaml files, it's pretty straight forward. The problem is that we need to make a decision on which files are taken for a 2014 historical simulation for the forcing files that require 2-year-after, do we use the ssp ones? which ones? do we use 2014 twice?

Reply:
This is a specific technical question and I documented this, its in release: [echam.datasets.yaml:L96-100](https://github.com/esm-tools/esm_tools/blob/release/configs/components/echam/echam.datasets.yaml#L96).

Reply:
Oh true, I did not remember that. So that situation is already handle with the current state of the scenarios. It seems that we are reached a conclusion: by default ESM-Tools will return an error if you try to run historical simulations of echam passed 2014, and the case of  ssp simulations will be the other way around. I'll do that tomorrow if no one complains against this solution. For FOCI people, who want to run simulations from the historical period into the ssp one, this restriction will be removed in the `foci.yaml` file.

Reply:
@PengyangSong, @chrisdane, @chrisdane, would this be an acceptable solution then?
![Screenshot 2022-11-25 at 19 04 03](https://user-images.githubusercontent.com/63242832/204037061-d1b3ef81-e183-4238-96ab-0f147f285843.png)

Reply:
Yeah ok. Maybe also print the initial date that raised the error.

Reply:
done!